client/src/lit/interactive-example/index.js

```diff
@@ -95,6 +119,92 @@ export class InteractiveExample extends GleanMixin(LitElement) {
     }
   }
 
+  /** @param {MouseEvent} event  */
+  _choiceClick({ target }) {
+    // TODO: use a different event handler for editor update event
+    // TODO: deal with update race conditions (editor updates after user clicks on different editor)
+    if (target instanceof PlayEditor) {
+      const choice = target.closest(".choice");
+      const choiceIndex = Array.prototype.indexOf.call(
+        choice?.parentNode?.children,
+        choice
+      );
+      this._selectChoice(choiceIndex, target.value);
+    }
+  }
+
+  _resetChoices() {
+    this.choiceSelected = -1;
+    this.choiceUnsupportedMask = 0;
+    const editorNodes = Array.from(
+      this.shadowRoot?.querySelectorAll("play-editor") || []
+    );
+    Array.from(editorNodes).forEach((editorNode, index) => {
+      editorNode.value = this._choices?.at(index) ?? "";
+    });
+    this._selectChoice(0);
+  }
+
+  /**
+   * @param {Number} index
+   * @param {string|undefined} code
+   */
+  async _selectChoice(index, code = undefined) {
+    code = typeof code === "string" ? code : this._choices?.at(index) || "";
+
+    if (!code) {
+      console.debug("No code selected");
+      return;
+    }
+
+    // TODO: nicer interface for posting messages than this:
+    const runner = this._runner.value;
+
+    if (!runner) {
+      console.error("No runner");
+      return;
+    }
+
+    // Ensures it has an iframe.
+    await runner.updateComplete;
+
+    const shadowRoot = runner.shadowRoot;
+
+    if (!shadowRoot) {
+      console.error("No shadowRoot");
+      return;
+    }
+
+    const iframe = shadowRoot.querySelector("iframe");
+
+    if (!iframe) {
+      console.error("No iframe");
+      return;
+    }
+
+    const contentWindow = iframe.contentWindow;
+
+    if (!contentWindow) {
+      console.error("No contentWindow");
+      return;
+    }
```

Comment:
All these checks are very verbose and not really necessary: what we should do is expose a method on `PlayRunner` which exposes `postMessage`

Reply:
Fixed via 253cd21a61.